dimse/dimse.go

```diff
@@ -109,17 +109,30 @@ func (d *messageDecoder) getString(tag dicomtag.Tag, optional isOptionalElement)
 	return v
 }
 
+// multiValueTags is a list of tags, that can hold multiple values (1-n)
+// https://dicom.nema.org/dicom/2013/output/chtml/part07/chapter_E.html
+var multiValueTags = map[dicomtag.Tag]bool{
+	dicomtag.OffendingElement:        true,
+	dicomtag.AttributeIdentifierList: true,
+}
+
 // Find an element with "tag", and extract a uint16 from it. Errors are reported in d.err.
 func (d *messageDecoder) getUInt16(tag dicomtag.Tag, optional isOptionalElement) uint16 {
 	e := d.findElement(tag, optional)
 	if e == nil {
 		return 0
 	}
-	v, err := e.GetUInt16()
+	if len(e.Value) > 1 && !multiValueTags[e.Tag] {
```

Comment:
I think to limit the scope of this workaround, maybe we should do it in the `decodeCMoveRsp` function: https://github.com/segmed/go-netdicom/blob/master/dimse/dimse_messages.go#L455

```go
	v.NumberOfRemainingSuboperations = d.getUInt16(dicomtag.NumberOfRemainingSuboperations, optionalElement)
```

--> get

```go
func (d *messageDecoder) getSuboperations(tag dicomtag.Tag, optional isOptionalElement) uint16 {
	// TODO: check if tag is one of
	// * NumberOfRemainingSuboperations
	// * NumberOfCompletedSuboperations
	// * NumberOfFailedSuboperations
	// * NumberOfWarningSuboperations

	e := d.findElement(tag, optional)
	if e == nil {
		return 0
	}
	v, err := e.GetUint16s()
	if err != nil {
		d.setError(err)
	}

	if len(v) > 0 {
		// some warning here
	}

	return v[0]
}

...

	v.NumberOfRemainingSuboperations = d.getSuboperations(dicomtag.NumberOfRemainingSuboperations, optionalElement)
```

so this workaround is only applied to NumberOf*Suboperatiosn elements, which is known to have this problem instead of everything with uint16 value.

Reply:
done